src/bolt/buffer.c

```diff
@@ -314,6 +360,48 @@ void buffer_write
 	buffer_index_add(buf, size);
 }
 
+// apply the mask to a single chunk in the buffer
+static void buffer_apply_mask_single
+(
+	buffer_index_t buf,    // buffer
+	uint32_t masking_key,  // masking key
+	uint32_t payload_len,  // payload length
+	int *offset            // in/out offset
+) {
+	char *payload = buf.buf->chunks[buf.chunk] + buf.offset;
+	int local_offset = *offset;
+	uint32_t double_mask[4] = {masking_key, masking_key, masking_key, masking_key};
+	uint64_t offset_mask = *(uint64_t *)((char *)double_mask + local_offset);
+	int i = 0;
+	for(; i + 8 <= payload_len; i+=8) {
+		*(uint64_t *)(payload + i) ^= offset_mask;
+	}
+	for(; i < payload_len; i++) {
```

Comment:
Why don't we go from 8 to 4 to 1 ?

Reply:
after we finish with 8 the would be max of 1 4 bytes so don't think is is worth it